src/main/src/main/java/org/geoserver/logging/GeoServerXMLConfiguration.java

```diff
@@ -0,0 +1,298 @@
+            ext = template.substring(template.lastIndexOf("."));
+        }
+
+        if (template.contains("-%")) {
+            // recognize formats similar to "logs/geoserver-%i.log"
+            ext = template.substring(template.indexOf("-%"));
+        } else if (template.contains("%")) {
+            // recognize formats similar to "logs/%hostName.log"
+            ext = template.substring(template.indexOf("%"));
+        }
+
+        if (template.contains("-$") && !ext.contains("-$")) {
+            // recognize formats similar to "logs/geoserver-$hostName-%i.log"
+            ext = template.substring(template.indexOf("-$"));
+        } else if (template.contains("$") && !ext.contains("$")) {
+            // recognize formats similar to "logs/$hostName-%i.log"
+            ext = template.substring(template.indexOf("$"));
+        }
+
+        if (Paths.extension(path) != null && ext.contains(".log")) {
+            // handles logs/geoserver-$i.log.gz
+            ext = ext.replace(".log", "." + Paths.extension(path));
+        }
+
+        if (path.lastIndexOf(".") == -1) {
+            return path + ext;
+        } else {
+            return path.substring(0, path.lastIndexOf(".")) + ext;
+        }
+    }
+
+    /** Post-process configuration. */
+    @Override
+    protected void doConfigure() {
+        super.doConfigure();
+
+        if (fileName() != null) {
+            LOGGER.debug("External logfileLocation provided:" + fileName());
+
+            // Confirm geoserverlogfile configured to use loggingLocation
+            Appender appender = getAppenders().get("geoserverlogfile");
+            if (appender instanceof RollingFileAppender) {
+                RollingFileAppender fileAppender = (RollingFileAppender) appender;
+                String fileName = fileAppender.getFileName();
+                String filePattern = fileAppender.getFilePattern();
+
+                LOGGER.debug("Postconfigure geoserverlogfile.filename=" + fileName);
+                LOGGER.debug("Postconfigure geoserverlogfile.filePattern=" + filePattern);
+            } else if (appender instanceof FileAppender) {
+                FileAppender fileAppender = (FileAppender) appender;
+                String fileName = fileAppender.getFileName();
+                LOGGER.debug("Postconfigure geoserverlogfile.filename=" + fileName);
+            }
+        }
+
+        if (suppressFileLogging) {
+            getAppenders().remove("geoserverlogfile");
+            for (LoggerConfig loggerConfig : getLoggers().values()) {
+                if (loggerConfig.getAppenders().containsKey("geoserverlogfile")) {
+                    LOGGER.debug(
+                            "Logger '"
+                                    + loggerConfig.getName()
+                                    + "' includes supressed 'geoserverlogfile':"
+                                    + loggerConfig.getAppenders().get("geoserverlogfile"));
+                }
```

Comment:
I will make the change to warning, however usually log4j logger here is set to error visibility only.
- I found that it is only test-cases that suppress file output.
- I kind of wish test cases used their own configurations rather than add so much complexity to the system.